java/src/jmh/java/com/mlt/BingMapsDecoderBenchmark.java

```diff
@@ -57,13 +59,67 @@ private void encodeTile(int z, int x, int y) throws IOException {
         y,
         encodedMvtTiles,
         encodedMvtTiles2,
+        compressedMVTiles,
         encodedMltTiles,
         tileMetadata,
         TestSettings.BING_MVT_PATH,
         SEPARATOR);
   }
 
   @Benchmark
+  public FeatureTable[] decodeMltZ4() {
+    var mlTile = encodedMltTiles.get(4);
+    var mltMetadata = tileMetadata.get(4);
+    return MltDecoder.decodeMlTileVectorized(mlTile, mltMetadata);
```

Comment:
The `FeatureTable` is the in-memory MLT format which i'm proposing as part of the [spec](https://github.com/maplibre/maplibre-tile-spec/tree/main/spec#2-in-memory-format). So basically the spec is divided into the storage and in-memory format like it is the case for example with Parquet and Arrow. This format can be directly processed like it is the case with the MVT in-memory representation but in a more efficient way. For example you can directly filter the data, process the geometries or in some cases directly copy the geometries into GL buffers. So the geometries are fully decoded and usable by (next next generation) map renderer. By not calling loadGeometry the (most) important and dominant part of the decoding is left out (such as transforming the command/parameter integer into vertices, also delta, zigZag, varint decoding). Even if the data are lazy decoded, in my experience (nearly) all geometries get materialized at some point in time before rendering (at least in data/styles  i'm using) and you pay the costs anyway. Or am I missing something in relation to the lazy decoding or do you have a different experience? 

But of course I also understand the problem that we cannot currently use this column-oriented in-memory format directly in MapLibre, as this would mean too much refactoring effort and we therefore have to convert to another row-oriented in-memory representation. What do think if we you have two benchmarks, one which shows the potential of the format by decoding in the proposed in-memory format which is representative for next generation map renderer (or also for map renderer which using different in-memory representations) and benchmarks for the currently used MapLibre GL JS in-memory representation. 
In general because of my research background i'm in particular interested in the "next generation" use case, so i'm also totally fine if we don't merge this benchmarks. On the other hand i also think just showing the second benchmarks where we have this additional transformation into a different in-memory representation doesn't fully show the potential of the format. What do you think about it?

Reply:
To be more detailed. The in-memory format allows map rendering libraries on the CPU as well on the GPU random (constant time) access to every attribute in the tile. The step of transforming to a random access representation is also currently quite expensive and has to be optimized.  So the MLT in-memory format has basically the same capabilities like the Arrow or Velox format but tailored for the map rendering use case. The GeoArrow format for example, which is basically a spatial extension to Arrow,  can also be directly processed and rendered by Deck.GL see https://github.com/geoarrow/deck.gl-layers